device.c

```diff
@@ -428,25 +453,52 @@ int adev_open_input_stream(struct audio_hw_device *dev,
         return -EINVAL;
     }
     /* check requested device */
-    /* TODO we should handle only devices described in XML configuration and "default" one */
     if ((devices & AUDIO_DEVICE_BIT_IN) == 0) {
         ALOGE("Failed. Incorrect device type. -EINVAL");
         *stream_in = NULL;
         pthread_mutex_unlock(&xdev->lock);
         return -EINVAL;
     }
 
-    /* TODO define card_id and dev_id for stream, using provided parameters:
-       devices, flags, address, source */
+    unsigned int slot = 0;
+    /* For now we have one card, so card id is always constant */
```

Comment:
add FIXME here

Reply:
Comment removed because card index was added to configuration table.